src/base/emu-i386/simx86/sigsegv.c

```diff
@@ -404,7 +406,9 @@ int e_handle_pagefault(dosaddr_t addr, unsigned err, sigcontext_t *scp)
 	/* We HAVE to invalidate all the code in the page
 	 * if the page is going to be unprotected */
 	addr &= _PAGE_MASK;
-	return InvalidateNodeRange(addr, PAGE_SIZE, p);
+	int ret = InvalidateNodeRange(addr, PAGE_SIZE, pjit);
+	if (TheCPU.err == EXCP_BREAKNODE) e_return_from_jit(scp, 1);
```

Comment:
I suspect this is something
from #2731 ?

Reply:
#2731 is related but independent: #2731 is specifically about when the last instruction in the block modifies the (same) block code; this PR is when any instruction modifies something (could be data or code) on the same page but cpatch isn't used, so the whole page is going to be unprotected.

Both can be true at the same time (last ins modifies, without cpatch) but that wasn't the case in #2708.